Modules/System/SharePoint/src/SharePointClientImpl.Codeunit.al

```diff
@@ -364,6 +364,7 @@ codeunit 9101 "SharePoint Client Impl."
         Request.Add('ContentTypesEnabled', true);
         Request.Add('Description', ListDescription);
         Request.Add('Title', ListTitle);
+        SharepointEvents.OnAfterAddListMetaData(Request);
```

Comment:
Adding events in the middle of methods is comparable to code customization. It's not a good pattern to follow. Also just consider that events are generally raised and taken out of context. You now need to recreate that context in a subscriber and figure out if you need to act. The more context is passed in the event the better but it will never be very good.

An alternative to this approach (as was suggested you look into) could be to change the methods so they take in a dictionary for example. You can then pass in all your data from the start and there's no need for events. It's a bit more work and you need to take care to not break the existing public methods but I think the end result will be much better.